# https://bugs.gentoo.org/666607
--- a/src/zm_db.cpp     2020-02-26 00:22:24.000000000 +0100
+++ b/src/zm_db.cpp     2020-02-26 11:42:05.856462044 +0100
@@ -40,7 +40,7 @@
     Error("Can't initialise database connection: %s", mysql_error(&dbconn));
     return false;
   }
-  my_bool reconnect = 1;
+  bool reconnect = 1;
   if ( mysql_options(&dbconn, MYSQL_OPT_RECONNECT, &reconnect) )
     Error("Can't set database auto reconnect option: %s", mysql_error(&dbconn));
   if ( !staticConfig.DB_SSL_CA_CERT.empty() )
