diff --git a/.gitignore b/.gitignore
index ee6ac35e7..905fde299 100644
--- a/.gitignore
+++ b/.gitignore
@@ -40,14 +40,14 @@ onvif/cmake_install.cmake
 onvif/modules/CMakeFiles/
 onvif/modules/MYMETA.json
 onvif/modules/MYMETA.yml
-onvif/modules/MakefilePerl
+onvif/modules/Makefile
 onvif/modules/cmake_install.cmake
 onvif/modules/output/
 onvif/modules/pm_to_blib
 onvif/proxy/CMakeFiles/
 onvif/proxy/MYMETA.json
 onvif/proxy/MYMETA.yml
-onvif/proxy/MakefilePerl
+onvif/proxy/Makefile
 onvif/proxy/cmake_install.cmake
 onvif/proxy/output/
 onvif/proxy/pm_to_blib
@@ -60,7 +60,7 @@ scripts/ZoneMinder/CMakeFiles/
 scripts/ZoneMinder/MYMETA.json
 scripts/ZoneMinder/MYMETA.yml
 scripts/ZoneMinder/Makefile.old
-scripts/ZoneMinder/MakefilePerl
+scripts/ZoneMinder/Makefile
 scripts/ZoneMinder/blib
 scripts/ZoneMinder/cmake_install.cmake
 scripts/ZoneMinder/lib/ZoneMinder/Base.pm
diff --git a/onvif/modules/CMakeLists.txt b/onvif/modules/CMakeLists.txt
index 502970af9..a4e32784d 100644
--- a/onvif/modules/CMakeLists.txt
+++ b/onvif/modules/CMakeLists.txt
@@ -9,10 +9,10 @@ endif()
 # MAKEMAKER_NOECHO_COMMAND previously defined in /scripts/zoneminder/CMakeLists.txt
 
 # Add build target for the perl modules
-add_custom_target(zmonvifmodules ALL perl Makefile.PL ${ZM_PERL_MM_PARMS_FULL} FIRST_MAKEFILE=MakefilePerl DESTDIR="${CMAKE_CURRENT_BINARY_DIR}/output" ${MAKEMAKER_NOECHO_COMMAND} COMMAND make -f MakefilePerl pure_install COMMENT "Building ZoneMinder perl ONVIF proxy module")
+add_custom_target(zmonvifmodules ALL perl Makefile.PL ${ZM_PERL_MM_PARMS_FULL} FIRST_MAKEFILE=Makefile DESTDIR="${CMAKE_CURRENT_BINARY_DIR}/output" ${MAKEMAKER_NOECHO_COMMAND} COMMAND make -f Makefile pure_install COMMENT "Building ZoneMinder perl ONVIF proxy module")
 
 # Add install target for the perl modules
 install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/output/" DESTINATION "/")
 
 # Add additional files and directories to make clean
-set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES "output;blib;pm_to_blib;MakefilePerl")
+set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES "output;blib;pm_to_blib;Makefile")
diff --git a/onvif/proxy/CMakeLists.txt b/onvif/proxy/CMakeLists.txt
index 779b4d47e..d993aa41a 100644
--- a/onvif/proxy/CMakeLists.txt
+++ b/onvif/proxy/CMakeLists.txt
@@ -9,10 +9,10 @@ endif()
 # MAKEMAKER_NOECHO_COMMAND previously defined in /scripts/zoneminder/CMakeLists.txt
 
 # Add build target for the perl modules
-add_custom_target(zmonvifproxy ALL perl Makefile.PL ${ZM_PERL_MM_PARMS_FULL} FIRST_MAKEFILE=MakefilePerl DESTDIR=${CMAKE_CURRENT_BINARY_DIR}/output ${MAKEMAKER_NOECHO_COMMAND} COMMAND make -f MakefilePerl pure_install COMMENT "Building ZoneMinder perl ONVIF proxy module")
+add_custom_target(zmonvifproxy ALL perl Makefile.PL ${ZM_PERL_MM_PARMS_FULL} FIRST_MAKEFILE=Makefile DESTDIR=${CMAKE_CURRENT_BINARY_DIR}/output ${MAKEMAKER_NOECHO_COMMAND} COMMAND make -f Makefile pure_install COMMENT "Building ZoneMinder perl ONVIF proxy module")
 
 # Add install target for the perl modules
 install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/output/" DESTINATION "/")
 
 # Add additional files and directories to make clean
-set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES "output;blib;pm_to_blib;MakefilePerl")
+set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES "output;blib;pm_to_blib;Makefile")
diff --git a/scripts/ZoneMinder/CMakeLists.txt b/scripts/ZoneMinder/CMakeLists.txt
index 2a8708d2f..c3cbb7ebb 100644
--- a/scripts/ZoneMinder/CMakeLists.txt
+++ b/scripts/ZoneMinder/CMakeLists.txt
@@ -25,10 +25,10 @@ else()
 endif()
 
 # Add build target for the perl modules
-add_custom_target(zmperlmodules ALL perl Makefile.PL ${ZM_PERL_MM_PARMS_FULL} FIRST_MAKEFILE=MakefilePerl DESTDIR="${CMAKE_CURRENT_BINARY_DIR}/output" ${MAKEMAKER_NOECHO_COMMAND} COMMAND make -f MakefilePerl pure_install COMMENT "Building ZoneMinder perl modules")
+add_custom_target(zmperlmodules ALL perl Makefile.PL ${ZM_PERL_MM_PARMS_FULL} FIRST_MAKEFILE=Makefile DESTDIR="${CMAKE_CURRENT_BINARY_DIR}/output" ${MAKEMAKER_NOECHO_COMMAND} COMMAND make -f Makefile pure_install COMMENT "Building ZoneMinder perl modules")
 
 # Add install target for the perl modules
 install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/output/" DESTINATION "/")
 
 # Add additional files and directories to make clean
-set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES "output;blib;pm_to_blib;MakefilePerl")
+set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES "output;blib;pm_to_blib;Makefile")
