# db.c:2034:19: warning: format ‘%s’ expects a matching ‘char *’ argument [-Wformat=]
diff --git a/srvrlib/db.c b/srvrlib/db.c
index 62c1432..da3abb3 100644
--- a/srvrlib/db.c
+++ b/srvrlib/db.c
@@ -2033,6 +2033,7 @@ get_db_max_rss(void)
 	} else if (db_min_mbyte > db_max_mbyte) {
 		quiet_trace_msg("ignore ./configure --with-db-memory=%d"
 				" that is > DB max=%s",
+				mbyte2str(db_min_mbyte),
 				mbyte2str(db_max_mbyte));
 		db_min_mbyte = DB_DEF_MIN_MBYTE;
 	} else if (complain) {
