diff --git a/manual/conf.py b/manual/conf.py
index f89364d4f..47dc8d0ae 100644
--- a/manual/conf.py
+++ b/manual/conf.py
@@ -19,7 +19,7 @@
 # import os
 # sys.path.insert(0, os.path.abspath('.'))
 
-import os, git, re, pathlib
+import os, re, pathlib
 
 # -- CAF-specific variables ---------------------------------------------------
 
@@ -37,26 +37,7 @@ with open(os.path.join(root_dir, "libcaf_core/caf/config.hpp")) as f:
     patch = raw_version % 100
     version = '{}.{}.{}'.format(major, minor, patch)
 
-# We're building a stable release if the last commit message is
-# "Change version to <version>".
-repo = git.Repo(root_dir)
-is_stable = False
-
-# We're building a stable release if this version has a release date.
-with open(os.path.join(root_dir, "CHANGELOG.md")) as f:
-    match = re.search('^## \[' + version + '\] - [0-9]{4}-[0-9]{2}-[0-9]{2}$',
-                      f.read(), re.MULTILINE)
-    if match != None:
-        is_stable = True
-
-# Generate the full version, including alpha/beta/rc tags. For stable releases,
-# this is always the same as the CAF version.
-if is_stable:
-    release = version
-else:
-    last_commit_full = str(repo.head.commit)
-    last_commit = last_commit_full[:7]
-    release = version + "-dev+exp.sha." + last_commit
+release = version
 
 # -- Enable Sphinx to find the literal includes -------------------------------
 
